apps/web/pages/availability/troubleshoot.tsx

```diff
@@ -13,40 +13,16 @@ type User = inferQueryOutput<"viewer.me">;
 
 const AvailabilityView = ({ user }: { user: User }) => {
   const { t } = useLocale();
-  const [loading, setLoading] = useState(true);
-  const [availability, setAvailability] = useState<{ end: string; start: string; title?: string }[]>([]);
   const [selectedDate, setSelectedDate] = useState(dayjs());
 
-  function convertMinsToHrsMins(mins: number) {
-    const h = Math.floor(mins / 60);
-    const m = mins % 60;
-    const hs = h < 10 ? "0" + h : h;
-    const ms = m < 10 ? "0" + m : m;
-    return `${hs}:${ms}`;
-  }
-
-  useEffect(() => {
-    const fetchAvailability = (date: Dayjs) => {
-      const dateFrom = date.startOf("day").utc().format();
-      const dateTo = date.endOf("day").utc().format();
-      setLoading(true);
-
-      fetch(`/api/availability/${user.username}?dateFrom=${dateFrom}&dateTo=${dateTo}`)
-        .then((res) => {
-          return res.json();
-        })
-        .then((availableIntervals) => {
-          setAvailability(availableIntervals.busy);
-        })
-        .catch((e) => {
-          console.error(e);
-        })
-        .finally(() => {
-          setLoading(false);
-        });
-    };
-    fetchAvailability(selectedDate);
-  }, [user.username, selectedDate]);
+  const { data, isLoading } = trpc.useQuery([
+    "viewer.availability.user",
+    {
+      user: user.username!,
+      dateFrom: selectedDate.startOf("day").utc().format(),
+      dateTo: selectedDate.endOf("day").utc().format(),
+    },
+  ]);
```

Comment:
Also we can be sure that we only fetch when the username is available

```suggestion
  const { data, isLoading } = trpc.useQuery(
    [
      "viewer.availability.user",
      {
        username: user.username!,
        dateFrom: selectedDate.startOf("day").utc().format(),
        dateTo: selectedDate.endOf("day").utc().format(),
      },
    ],
    { enabled: !!user.username }
  );
```